lib/spack/spack/installer.py

```diff
@@ -2148,20 +2201,42 @@ def install(self) -> None:
             if ltype == "read":
                 lock.release_read()
                 self._requeue_task(task, install_status)
-                continue
+                return None
 
             # Proceed with the installation since we have an exclusive write
             # lock on the package.
-            install_status.set_term_title(f"Installing {pkg.name}")
+            # Start a child process for a task that's ready to be installed.
+
+            install_status.set_term_title(f"Installing {task.pkg.name}")
             try:
                 action = self._install_action(task)
 
                 if action == InstallAction.INSTALL:
-                    self._install_task(task, install_status)
+                    try:
+                        task.start()
+                        tty.msg(install_msg(pkg_id, self.pid, install_status))
+                        active_tasks.append(task)
+                    except Exception as e:
+                        print(f"Error occurred during InstallAction.INSTALL: {e}")
                 elif action == InstallAction.OVERWRITE:
                     # spack.store.STORE.db is not really a Database object, but a small
                     # wrapper -- silence mypy
-                    OverwriteInstall(self, spack.store.STORE.db, task, install_status).install()  # type: ignore[arg-type] # noqa: E501
+                    OverwriteInstall(
+                        self, spack.store.STORE.db, task, install_status
+                    ).install()  # type: ignore[arg-type] # noqa: E501
+            except Exception as e:
+                print(f"Error occurred while determining the install action: {e}")
+            return task
+
+        def complete_task(task):
```

Comment:
```suggestion
        def complete_task(task) -> None:
```

I think this doesn't need to return anything -- since we don't use the return value.